Raven.Database/Bundles/Replication/Tasks/ReplicationTask.cs

```diff
@@ -256,24 +256,32 @@ public Task ExecuteReplicationOnce(bool runningBecauseOfDataModifications)
                     {
                         if (log.IsDebugEnabled)
                             log.Debug("Replication to distination {0} skipped due to existing replication operation", dest.ConnectionStringOptions.Url);
+                        //Although we set the event when the replication task is done we don't take into consideration the filtered documents.
+                        docDb.WorkContext.ReplicationResetEvent.Set();
```

Comment:
This is what you told me to implement i wanted to just take care of
filtered documents and avoid this and the wait that follows.
It will not cause us to always run since we will be stuck waiting for one
of the task to complete.
currently the code in 3.0 has this behavior where you would run again and
again and again the wait for work block under even a slow stream of work
will always signal.

On Mon, Dec 21, 2015 at 10:11 AM, Ayende Rahien <notifications@github.com>
wrote:

> In Raven.Database/Bundles/Replication/Tasks/ReplicationTask.cs
> <https://github.com/ayende/ravendb/pull/3030#discussion_r48124045>:
>
> > @@ -256,24 +256,32 @@ public Task ExecuteReplicationOnce(bool runningBecauseOfDataModifications)
> >                      {
> >                          if (log.IsDebugEnabled)
> >                              log.Debug("Replication to distination {0} skipped due to existing replication operation", dest.ConnectionStringOptions.Url);
> > +                        //Although we set the event when the replication task is done we don't take into consideration the filtered documents.
> > +                        docDb.WorkContext.ReplicationResetEvent.Set();
>
> Won't this cause us to always run? Instead of waiting when we have no work
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ayende/ravendb/pull/3030/files#r48124045>.
>



-- 

*Hibernating Rhinos Ltd  *[image: cid:image001.png@01CF95E2.8ED1B7D0]

Tal Weiss* l **Core Team Developer l *Mobile:+972-54-802-4849

Office: +972-4-622-7811* l *Fax: +972-153-4-622-7811*l* Skype: talweiss1982

*Raven**DB** paving the way to "Data Made Simple" http://ravendb.net/
<http://ravendb.net/> *